File: dynamic/src/main/java/com/github/rviannaoliveira/dynamic/domain/model/DynamicAnalyticsProperties.kt
package com.github.rviannaoliveira.dynamic.domain.model

import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import com.fasterxml.jackson.annotation.JsonProperty

@JsonIgnoreProperties(ignoreUnknown = true)
data class DynamicAnalyticsProperties(
    @JsonProperty("category")
    val category: String,

    @JsonProperty("action")
    val action: String,

    @JsonProperty("label")
    val label: String,

    @JsonProperty("track")
    val track: String,
)

File: dynamic-flutter/flutter_dynamics_module/lib/data/model/action/dynamic_analytics_properties.dart
class DynamicAnalyticsProperties {
  final String? category;
  final String? action;
  final String? label;
  final String? track;

  const DynamicAnalyticsProperties(
      {this.category, this.label, this.action, this.track});

  factory DynamicAnalyticsProperties.fromJson(dynamic properties) {
    return DynamicAnalyticsProperties(
      category: properties != null ? properties["category"] : null,
      action: properties != null ? properties["action"] : null,
      label: properties != null ? properties["label"] : null,
      track: properties != null ? properties["track"] : null,
    );
  }

  static String key = "analytics";
}
